code/modules/client/client_procs.dm

```diff
@@ -260,6 +260,8 @@ GLOBAL_VAR_INIT(external_rsc_url, TRUE)
 	if(isliving(mob))
 		var/mob/living/L = mob
 		L.away_time = world.time
+		if (isxeno(L))
```

Comment:
make a mob/living/proc/begin_away() and override it for xenos to do the specific behaviour.

Reply:
like
```
/mob/living/proc/begin_away()
 return

/mob/living/carbon/Xenomorph/begin_away()
 addtimer(CALLBACK(src, .proc/handle_afk_takeover), XENO_AFK_TIMER)
```